Pocket/pocket-ios
# Shared With You

## Summary
* This PR adds the [Shared With You feature](https://developer.apple.com/wwdc22/10094) to Pocket
## References 
* POCKET-9636

## Implementation Details
* Add `SharedWithYouStore`, a store for `SWHighlight` objects that correspond to items that were shared with you via Messages
* Add a Shared With You carousel section with cells displaying the attribution view at the bottom
* Add a Shared With You detail list that appears when tapping the "See All" button on the Shared With You section
* Add a `SharedWithYouItem` entity to Core Data
* Add GraphQL queries for items associated to Shared With You URLs
> [!NOTE]
> The following will be added in separate PRs
>  - Analytics 
>  - More testing 
>  - Support for Pocket.co

> [!WARNING]
> This PR contains a few user-generated warnings as placeholders for analytics. These will be all resolved before shipping the feature

> [!IMPORTANT]
> Thank you @bassrock ! Your shared with you [prototype](#234) has been of great help for this one

## Test Steps
#### Pre-requisites:  ####
- build/run on a device where you have messages containing links of collections or syndicated articles shared from Pocket by another user 
- make sure to enable shared with you in your iOS device Settings:
    - Tap the "Settings" icon on your iPhone or iPad
    - Scroll down to find the "Messages" app, then tap on it
    - Scroll down to find the "Shared With You" button, then tap on it
    - make sure the "Automatic Sharing" setting is turned on
    - Scroll down to find "Pocket" in the app list, and make sure the toggle is turned on.

#### Test steps: ####

* Build/run and make sure you have shared Pocket items (see pre-requisite) in your Messages app
* Verify that you see the Shared With You section in Home. This will contain a maximum of 4 items. All the other items will be available when tapping on "See All"
* Verify that you can save/unsave and share any item
* Verify that tapping on the attribution view does one of the following
    - if the item was shared by only one user, opens the Messages app with an opportunity to reply to that message containing the item
    - if the item was shared by more than one users, you will be given the option to select an user and then execute the previous step
- Verify that tapping on "See All" in the Shared With You section takes you to the list of all the items
- Verify that the standard actions work as expected on those items

## PR Checklist:
- [ ] ~~Added Unit / UI tests~~ future PR

## Screenshots

<p align=center>

<img src=https://github.com/Pocket/pocket-ios/assets/34376330/6372b644-cd21-48a7-a1a4-7e86468497a9>


</p>
